app/js/endTest.ts

```diff
@@ -1,10 +1,14 @@
+let count: number = 0;
+
 function pageLeaveAlert() {
-    alert ('Beware, if you leave the page the test will end!')
+    if (count < 2) {
```

Comment:
This works well - however I still occasionally find that I get the alert, and then instantly get another one upon closing it.

Could you add something to limit how often the popups can appear? The suggestion in the bug report is 120 seconds, maybe only allow it once every 2 minutes along side limiting how many times it can display in total